_source/_posts/2023-07-27-spring-graphql-react.md

```diff
@@ -0,0 +1,1794 @@
+    healthcheck:
+      test: ['CMD', 'wget', 'http://localhost:7474/', '-O', '-']
+      interval: 5s
+      timeout: 5s
+      retries: 10
+```
+
+Create the file `src/main/docker/.env` with the following content:
+
+__.env__
+```shell
+NEO4J_PASSWORD=verysecret
+```
+
+As you can see the compose file will mount `<csv-folder>` to a `/var/lib/neo4j/import` volume, making the content accessible from the running neo4j container.
+Replace `<csv-folder>` with the path to the CSV files downloaded before.
+
+In a terminal, go to the `docker` folder and run:
+
+```shell
+docker compose -f neo4j.yml up
+```
+
+### Run the API server
+
+Go to the project root folder and start the application with:
+
+```shell
+source .env && ./gradlew bootRun
+```
+
+Wait for the logs to inform the seed data migrations have run (it might take a while):
+
+```
+2023-08-02T13:06:14.386-03:00  INFO 28673 --- [           main] a.s.neo4j.migrations.core.Migrations     : Applied migration 001 ("Constraint").
+2023-08-02T13:06:23.379-03:00  INFO 28673 --- [           main] a.s.neo4j.migrations.core.Migrations     : Applied migration 002 ("Company").
+2023-08-02T13:11:23.693-03:00  INFO 28673 --- [           main] a.s.neo4j.migrations.core.Migrations     : Applied migration 003 ("Person").
+2023-08-02T13:21:03.680-03:00  INFO 28673 --- [           main] a.s.neo4j.migrations.core.Migrations     : Applied migration 004 ("PersonCompany").
+2023-08-02T13:21:06.519-03:00  INFO 28673 --- [           main] a.s.neo4j.migrations.core.Migrations     : Applied migration 005 ("CompanyData").
+2023-08-02T13:21:06.551-03:00  INFO 28673 --- [           main] a.s.neo4j.migrations.core.Migrations     : Applied migration 006 ("Land").
+```
+
+Test the API with GraphiQL at **http://localhost:8080/graphiql**. In the query box on the left, paste the following query
+
+```graphql
+{
+    companyList(page: 20) {
+        id
+        SIC
+        name
+        status
+        category
+        companyNumber
+        countryOfOrigin
+    }
+}
+```
+
+You should see the query output in the box on the right:
+
+{% img blog/spring-graphql-react/graphiql-test.png alt:"GraphiQL example" width:"900" %}{: .center-image }
+
+
+> __NOTE:__
+> If you see a warning message in the server logs, that reads _The query used a deprecated function: id_, you can ignore it, Spring Data Neo4j still [behaves correctly](https://github.com/spring-projects/spring-data-neo4j/issues/2716).
+
+
+## Build a React client
+
+Now let's create a Single Page Application (SPA) to consume the GraphQL API with React and Next.js. The list of companies will display in a [MUI](https://mui.com/material-ui/getting-started/) [Datagrid](https://mui.com/x/react-data-grid/) component. The application will use Next.js `app` router. The `src/app` folder will contain routing files only, and the UI components and application code will be located in other folders.
+
+Install Node and in a terminal run:
+
+```shell
+npx create-next-app@13.4.19
```

Comment:
Should I run this in the root of the Spring Boot app or a directory alongside it? I could look at the GitHub repo to figure it out, but I think you should say it here to reduce ambiguity.

Reply:
Added a comment.